pyspark/dl/dataset/transformer.py

```diff
@@ -24,3 +26,86 @@ def normalizer(mean, std):
     """
     return lambda sample: Sample.from_ndarray((sample.features - mean) / std,
                                               sample.label, sample.bigdl_type)
+
+
+def pixel_normalizer(mean):
+    """
+    Normalize with pixel mean array
+    :param mean: mean ndarray for each pixel
+    :return:
+    """
+    return lambda sample: Sample((sample.features - mean),
+                                              sample.label, sample.bigdl_type)
+
+
+def crop(crop_width, crop_height, crop_method='random'):
+    """
+    Crop image sample to specified width and height
+    :param crop_width: width after cropped
+    :param crop_height: height after cropped
+    :param crop_method: crop method, should be random or center
+    :return: cropped image sample
+    """
+    def func(sample):
+        h = sample.features.shape[0]
+        w = sample.features.shape[1]
+        if crop_method == 'random':
+            x1 = random.randint(0, w - crop_width)
+            y1 = random.randint(0, h - crop_height)
+        elif crop_method == 'center':
+            x1 = (w - crop_width)/2
+            y1 = (h - crop_height)/2
+        cropped = sample.features[y1:y1+crop_height, x1:x1+crop_width]
+        return Sample(cropped, sample.label, sample.bigdl_type)
+    return func
+
+
+def channel_normalizer(mean_r, mean_g, mean_b, std_r, std_g, std_b):
+    """
+    Normalize image sample by means and std of each channel
+    :param mean_r: mean for red channel
+    :param mean_g: mean for green channel
+    :param mean_b: mean for blue channel
+    :param std_r: std for red channel
+    :param std_g: std for green channel
+    :param std_b: std for blue channel
+    :return: normalized image sample
+    """
+    def func(sample):
```

Comment:
No need to operate on a sample. We should pass in an image with a specific format. i.e. PIL.Image which supported by ``` Pillow ```
``` Pillow ``` can offer us lots of useful Image loading and transforming operations and also support Python3.
OpenCV can also be another option and we can support it in other modules in the future.